### DISCUSSION TITLE
The editor doesn't export/download the scene after adding variants to the model.

### DISCUSSION THREAD

**Discussion by plutorkhan** [google/model-viewer#3746] [Q&A]
I want to apply different texture variants to a hottub 3d model via model-viewer editor. I could import and add textures per variant, but couldn't download the scene. I've tested other glb 3d models the same bug happens again.
The console logs:

 (in promise) TypeError: Cannot read properties of undefined (reading 'material')
    at space-opera.js:88772:60
    at Mesh.traverse (space-opera.js:33845:3)
    at Group.traverse (space-opera.js:33851:18)
    at GLTFExporterMaterialsVariantsExtension.beforeParse (space-opera.js:88763:20)
    at space-opera.js:88132:27
    at GLTFWriter._invokeAll (space-opera.js:88178:4)
    at GLTFWriter.processInput (space-opera.js:88130:8)
    at GLTFWriter.write (space-opera.js:86499:8)
    at GLTFExporter.parse (space-opera.js:86187:10)
    at space-opera.js:90230:26

I used chrome and edge, the results are the same.

**Comment by elalish** [comment]
It's working for me; can you please list specific repro steps? Something like
1) load this model - <link to model download>
2) click X button
3) click y button
...
n) click download scene

**Comment by jgboulay** [comment]
I had this issue as well. To get around it I had to create the variants in the model editor and import each material as I created a variant. Our Blender developer suggested that materials baked into the model may be to large for the 'download scene' to handle.

**Comment by elalish** [comment]
Perhaps, but I can tell you exactly what's going on if you can give me reliable repro steps.

**Comment by Readerr57** [comment]
I have same problem. I have tested Mac and Windows with different browsers.

Here is how it happens:

1. Go to editor
2. Select Model: Astronaut
3. Create two material variants
4. Try to "Download scene" button. Nothing happens.

Button color changes to light blue when clicked, but nothing happens. Waiting doesn`t help.

**Reply by elalish** [reply]
Thanks, I can repro now: #3814

**Comment by Readerr57** [comment]
I found a workaround for this. See more details here: https://github.com/google/model-viewer/issues/3814

You need to edit each material variant, so that variant name appears after selected material name. When you have edited each material on each variant, you can download model normally.
